File: src/main/java/io/nightfis/createstellarexoduscore/filter/IsPlayerFilter.java
package io.nightfis.createstellarexoduscore.filter;

import net.minecraft.network.chat.Component;
import net.minecraft.world.entity.LivingEntity;
import net.minecraft.world.entity.player.Player;

public class IsPlayerFilter extends TargetFilter {

    private static final String ID = "IsPlayerFilter";

    @Override
    public String getId() {
        return ID;
    }

    @Override
    public boolean match(LivingEntity entity, String arg) {
        return entity instanceof Player;
    }

    @Override
    public Component title() {
        return Component.translatable("screen.createstellarexoduscore.auto_aim_turret.filter.is_player");
    }
}

File: src/main/kotlin/io/nightfis/createstellarexoduscore/filter/IsPlayerFilter.kt
package io.nightfis.createstellarexoduscore.filter

import net.minecraft.network.chat.Component
import net.minecraft.world.entity.LivingEntity
import net.minecraft.world.entity.player.Player

class IsPlayerFilter : TargetFilter() {

    override fun getId(): String = ID

    override fun match(entity: LivingEntity, arg: String): Boolean = entity is Player

    override fun title(): Component {
        return Component.translatable("screen.create_stellar_exodus_core.auto_aim_turret.filter.is_player")
    }

    companion object {
        private const val ID = "IsPlayerFilter"
    }
}
